### PR TITLE
QtLocationPlugin: Remove Unused Geocoding

### PR DESCRIPTION
**Pull request by HTRamsey**
Remove Unused Geocoding manager, as I don't believe it has ever been used. Removed old Qt license as it has changed.

### CODE REVIEW

**Comment by HTRamsey** [comment]
@bkueng This one is pretty straightforward, these files were written like 9 years ago and were never used.

File: src/QtLocationPlugin/QGeoServiceProviderPluginQGC.cpp [hunk 1]
```diff
@@ -1,96 +1,58 @@
...
 #endif
-//-----------------------------------------------------------------------------
-QGeoCodingManagerEngine*
-QGeoServiceProviderFactoryQGC::createGeocodingManagerEngine(
+
+QGeoCodingManagerEngine* QGeoServiceProviderFactoryQGC::createGeocodingManagerEngine(
     const QVariantMap &parameters, QGeoServiceProvider::Error *error, QString *errorString) const
 {
-    return new QGeoCodingManagerEngineQGC(parameters, error, errorString);
+    *error = QGeoServiceProvider::NotSupportedError;
```
Comment (bkueng) [inline src/QtLocationPlugin/QGeoServiceProviderPluginQGC.cpp:23]: According to the [docs](https://doc.qt.io/qt-6/qgeoserviceproviderfactory.html#createGeocodingManagerEngine), `error` might be nullptr

**Review by bkueng — COMMENTED** [review]
Nice, removing code

File: src/QtLocationPlugin/QGeoServiceProviderPluginQGC.cpp (hunk 1 above)
Comment (HTRamsey) [reply src/QtLocationPlugin/QGeoServiceProviderPluginQGC.cpp:23]: Huh, wonder if they never updated it from Qt5. I checked the QtLocation code and it's impossible to be nullptr but I can add checks anyways

**Review by bkueng — APPROVED** [review]

**Comment by HTRamsey** [comment]
sorry just converted my tabs to spaces

**Pull request merged**